File: Algorithms_4th_Edition/c/1/3/book/stats.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "Bag.h"

int main(void)
{
    Bag numbers;
    double num,sum,std,mean;
    initBag(&numbers);
    while(scanf("%lf",&num) == 1)
        add(&numbers,num);
    sum = 0.0;
    
    Node * h;
    h = numbers.items;
    while( h != NULL)
    {
        sum += h -> item;
        h = h -> next;
    }
    int n = size(&numbers);
    mean = sum / n;
    h = numbers.items;
    sum = 0.0;
    while(h != NULL)
    {
        sum += (h -> item - mean) * (h->item - mean);
        h = h -> next;
    }
    std = sqrt(sum / (n-1));
    printf("Mean: %.2f\n",mean);
    printf("Std dev: %.2f\n",std);
    return 0;
}

File: Algorithms_4th_Edition/java/src/1/3/book/Stats.java
import edu.princeton.cs.algs4.Bag;
import edu.princeton.cs.algs4.StdIn;

public class Stats {
    public static void main(String[] args) {
        Bag<Double> numbers = new Bag<Double>();

        while (!StdIn.isEmpty())
            numbers.add(StdIn.readDouble());

        int N = numbers.size();
        double sum = 0.0;
        for (Double x : numbers)
            sum += x;
        double mean = sum / N;
        sum = 0.0;
        for(Double x : numbers)
            sum += (x - mean) * (x - mean);
        double std = Math.sqrt(sum / (N - 1));
        System.out.printf("Mean: %.2f\n",mean);
        System.out.printf("Std dev: %.2f\n",std);
    }
}
